### PR TITLE
Bump wagon dependency to latest version.

### LINKED ISSUE #2443: Please upgrade libwagon to version 3.0.0
As reported in the dependency https://github.com/cemerick/pomegranate/issues/101, upstream maven is using wagon 3.0.0 and version 2.12 is no longer supported.

Hence, we should upgrade when possible.

### PR DESCRIPTION
**Pull request by technomancy**
@ehashman What do you think about bumping this in Leiningen instead since https://github.com/cemerick/pomegranate/pull/103 isn't getting merged?

Fixes #2443

### CODE REVIEW

**Comment by ehashman** [comment]
Oh, you may also want to bump Maven's version.

**Comment by ehashman** [comment]
Hmmm, that looks broken... poor pomegranate. Try maven 3.5.3? I got that one to work in my PR to pomegranate.

File: leiningen-core/project.clj [hunk 1]
```diff
@@ -9,11 +9,21 @@
                  [org.flatland/classlojure "0.7.1"]
                  [robert/hooke "1.3.0"]
                  [com.cemerick/pomegranate "1.0.0"
-                  :exclusions [org.codehaus.plexus/plexus-utils]]
+                  :exclusions [org.codehaus.plexus/plexus-utils
+                               org.apache.maven.wagon/wagon-provider-api
+                               org.apache.maven.wagon/wagon-http
+                               org.apache.httpcomponents/httpclient
```
Comment (ehashman) [inline leiningen-core/project.clj]: I wonder if this exclusion is the problem with the build?

**Comment by ehashman** [comment]
Pomegranate 1.1.0 is now available \o/

**Comment by technomancy** [comment]
Fixed in dac2ede instead.

**Pull request closed without merging**